### DISCUSSION TITLE
What is the recommended way to upgrade CRDs?

### DISCUSSION THREAD

**Discussion by michaelgeorgeattard** [argoproj/argo-cd#5160] [Q&A]
I installed Argo CD using Helm 3, and when upgrading the chart the CRDs are not touched.

What is the best way to upgrade the CRDs to the latest version defined in the Helm chart, preferably without incurring downtime?

**Comment by sbose78** [comment]
Hi @michaelgeorgeattard !

I'm assuming that you have placed the CRDs in the `/crds` directory of your chart. With such a layout, Helm 3 skips any upgrades to the CRD as per design.

In order to have Helm 3 upgrade charts for you, you would need to have two charts:

1. The CRD definition in one chart.
2. Any resources that use that CRD in another chart.

After that's done, you may have their sync'd using ArgoCD. 

Optionally, you may also need to take care of breaking changes which may affect existing CRs. You may want to consider use of [resource hooks](https://argoproj.github.io/argo-cd/user-guide/resource_hooks/) to take migrate CRs to a the newer version if needed.

**Comment by michaelgeorgeattard** [comment]
Hi @sbose78

First of all thanks for your answer.

To clarify, I am referring to the Argo CD Helm chart https://github.com/argoproj/argo-helm/tree/master/charts/argo-cd.

Can I just manually apply the newest CRDs for `AppProject` and `Application` without affecting the current CRs?

**Reply by sbose78** [reply]
> Can I just manually apply the newest CRDs for AppProject and Application without affecting the current CRs?

Yes.
In principle, as long as the new v/s old  `AppProject` and `Application` CRD doesn't have any breaking changes which affect the CRs in your cluster, you are good.

Which version are you upgrading from & which version are you upgrading to?

**Reply by michaelgeorgeattard** [reply]
Hi @sbose78 

I am upgrading from `1.4.2` to `1.7.2`.

Is there a changelog or documentation I can go through to see if the CRDs have undergone breaking changes across those two versions?
